lib/cocoapods/installer/analyzer.rb

```diff
@@ -449,9 +451,14 @@ def generate_target(target_definition, pod_targets)
       # @return [Boolean] if the pod target is only referenced from test dependencies.
       #
       def pod_target_test_only?(pod_target, pod_targets)
-        source = pod_targets.any? { |pt| pt.dependent_targets.map(&:name).include?(pod_target.name) }
-        test = pod_targets.any? { |pt| pt.test_dependent_targets.map(&:name).include?(pod_target.name) }
-        !source && test
+        name = pod_target.name
+        key = [name, pod_targets]
```

Comment:
Depending on how often this is hit, it might be worth using a struct as the key instead of an array